Perun

choeppler opened the pull request:
Blockchain-agnostic State Channels framework, written in Go.

choeppler commented:
@lehors, could you confirm sponsorship, please ?

tkuhrt commented on labs/direct-state-transfer_perun.md:
```diff
@@ -0,0 +1,62 @@
+
+- <https://github.com/lehors> - Chair of the TSC
+
+# Pre-existing repository
```
At least the go-perun repository does not have DCO signoffs on all commits. DCO signoffs will be required if we were to transfer the repos to Hyperledger Labs. Can you comment on your expectation for transferring these repos, as well as, whether you would want five separate repos? Or would you only like a single repository that you will create sub-directories under and manually bring the code from these repos into?

choeppler replied:
The [contracts-eth](https://github.com/direct-state-transfer/contracts-eth) repository is currently in the process of being re-licensed to Apache-2.0.

As for the Signed-off-by-requirement: the three [dst-* repositories](https://github.com/direct-state-transfer?q=dst-d&type=&language=) should be fine, and we planned to add the signoffs to [go-perun](https://github.com/direct-state-transfer/go-perun) and [contracts-eth](https://github.com/direct-state-transfer/contracts-eth) as soon as the proposal gets accepted so all of the repositories can be transferred. Is that okay?

And yes, we'd like to keep the separate repositories.

tkuhrt replied:
Licensing is important. All code will need to be Apache 2.0.

As long as signoffs exist before the transfer, that would be acceptable.

sebastianst replied:
@tkuhrt Would it be ok to leave the SDK at its current name `go-perun` and prefix all other repos just with `perun-`? It is custom for Go repos/projects to name them `go-...`, like `go-ethereum` etc. so we'd prefer to leave the SDK as `go-perun`. This way we also wouldn't need to rename it in several places where it is already used. The other repos can start with `perun-...`. Also, `perun-framework-` would be a very long and cumbersome prefix so `perun-` would be preferable. E.g. `perun-contracts-eth` is already long but `perun-framework-contracts-eth` is not nice IMO.

tkuhrt commented on labs/direct-state-transfer_perun.md:
```diff
@@ -0,0 +1,62 @@
+# Lab Name
```
Are you looking for the lab name (and thus repo) to be named direct-state-transfer, perun, or direct-state-transfer-perun? Based on what you want, please make sure that the .md file matches your expectation.

choeppler replied:
Ok. I am going to discuss this with all involved parties and then change the md file accordingly.

choeppler replied:
I updated the lab name to "Perun Framework" and changed this PR's title and md file accordingly. Cf. 3c8a796

tkuhrt commented on labs/direct-state-transfer_perun.md:
```diff
@@ -0,0 +1,62 @@
+
+# Scope of Lab
+
+Perun / Direct State Transfer is a Go implementation of the [Perun state
```
What blockchain platforms is this lab looking to support?

choeppler replied:
We try to build the software blockchain agnostic, with the ledger specific parts being implemented in different backends. Currently the implementation supports Ethereum and our focus of development is to further implement the Perun state channel protocol and node. Once that's in place, we'd certainly want to ensure it works with Besu and support other blockchains depending on community interest  / feedback.

Of course, if there is immediate interest of someone wanting to contribute support for another blockchain, we'd also be happy to help with implementing the corresponding blockchain backend.

sebastianst commented on labs/perun-framework.md:
```diff
@@ -1,14 +1,14 @@
 # Scope of Lab
 
-Perun / Direct State Transfer is a Go implementation of the [Perun state
+Perun Framework is a Go implementation of the [Perun state
```
```suggestion
The Perun Framework is a suite of implementations around the [Perun state
```
I would not mention implementation details like Go in the general description of the Perun framework. For example, The Ethereum smart contracts are written in Solidity, its tests are written in TypeScript etc. In the future, we might also add additional implementations in other languages.

choeppler replied:
@sebastianst, thank you for the review! Your suggested change makes sense. I've added it in commit 3a1a883.

sebastianst commented on labs/perun-framework.md:
```diff
@@ -1,14 +1,14 @@
 
 # Short Description
 
 Blockchain-agnostic State Channels framework, written in Go.
```
```suggestion
Blockchain-agnostic State Channels framework.
```
see other suggestion for comment

choeppler replied:
Changed in commit 3a1a883.

lehors commented:
> @lehors, could you confirm sponsorship, please ?

I do.

choeppler commented:
@tkuhrt and @lehors, thank you very much for the positive review! Looking forward to join Hyperledger Labs soon :-)

Regarding the remaining [open issues](https://github.com/hyperledger-labs/hyperledger-labs.github.io/pull/136#pullrequestreview-446576681):
* [ ] We'll address the licensing and sign-offs issues in the coming days.
* [ ] As our repos should be prefixed with the lab's name and @sebastianst suggested to just use `perun-` as prefix in a [comment above](https://github.com/hyperledger-labs/hyperledger-labs.github.io/pull/136#discussion_r452980220), I propose to rename the Lab to "Perun" to settle that. Okay? Then I'll add a commit to this PR renaming the md-file accordingly.

Two additional questions:
1. Do you want us to rename the repos and update the contributing guides to reflect the DCO requirement before they are moved or as a first action after moving to the Hyperledger Labs org?
2. Would you like me to squash the commits of this PR so you have a cleaner commit history after the merge?

tkuhrt commented:
> * [ ]  As our repos should be prefixed with the lab's name and @sebastianst suggested to just use `perun-` as prefix in a [comment above](https://github.com/hyperledger-labs/hyperledger-labs.github.io/pull/136#discussion_r452980220), I propose to rename the Lab to "Perun" to settle that. Okay? Then I'll add a commit to this PR renaming the md-file accordingly.
> 
I am okay with renaming the lab to `perun`.

> 1. Do you want us to rename the repos and update the contributing guides to reflect the DCO requirement before they are moved or as a first action after moving to the Hyperledger Labs org?
It would be great to have the repos renamed prior to transferring to the Hyperledger Labs org.

> 2. Would you like me to squash the commits of this PR so you have a cleaner commit history after the merge?
We can always squash when we merge, but we are happy to have you do that.

sebastianst commented on labs/perun-framework.md:
```diff
@@ -0,0 +1,62 @@
+
+# Scope of Lab
+
+The Perun Framework is suite of implementations around the [Perun state
```
```suggestion
The Perun Framework is a suite of implementations around the [Perun state
```
typo, missing article

choeppler replied:
fixed in 484ffbe

sebastianst commented:
> I am okay with renaming the lab to `perun`.

Thanks! As mentioned before, we'd prefer to leave the name of the SDK as `go-perun`, as it is already used in several places etc. Is that possible? @tkuhrt

sebastianst commented:
> > I am okay with renaming the lab to `perun`.
> 
> Thanks! As mentioned before, we'd prefer to leave the name of the SDK as `go-perun`, as it is already used in several places etc. Is that possible? @tkuhrt

Bumping this question @tkuhrt would really be great if we can keep the name `go-perun` for the sdk. Thanks.

tkuhrt commented:
> > > I am okay with renaming the lab to `perun`.
> > 
> > 
> > Thanks! As mentioned before, we'd prefer to leave the name of the SDK as `go-perun`, as it is already used in several places etc. Is that possible? @tkuhrt
> 
> Bumping this question @tkuhrt would really be great if we can keep the name `go-perun` for the sdk. Thanks.

My bad. If this is the convention, then I guess we go with it (no pun intended). :)

Let us know when you have made it through the process of renaming, license updates, and DCO signoffs

choeppler commented:
@tkuhrt, we are finally done with the process renaming, license updates and DCO signoffs. Where necessary, I've also prepared updates for the Contributing guides to reflect the change in the legal framework from the CLA used previously to the DCO used in Hyperledger Labs. See the PRs referenced above.

Finally, I've rebased this PR on master and squashed all commits. So we should be ready to merge and move the repos into the Hyperledger-Labs org. :-)

Please advise on how to move forward.

tkuhrt commented:
> Please advise on how to move forward.

@choeppler : If you transfer the repositories to me (tkuhrt), then I will transfer them to the Hyperledger Labs organization.

choeppler commented:
@tkurt : directly transferring a repository to you from the direct-state-transfer org didn't work:
```
You can only transfer a repository from an organization to yourself at this time 
```
So I'll try to transfer them to you via me (choeppler). Just started with the first one. If that works, I'll send the others.

tkuhrt commented:
@choeppler : I was able to successfully transfer go-perun to the hyperledger-labs organization. I have added the perun-committers team that the committers of the lab will need to accept their invitations for access. Please feel free to transfer the other repos, and I will get them transferred over too.

choeppler commented:
@tkuhrt : Great! I've just transferred the other four repos.

tkuhrt commented:
All repos transferred. Merging this PR.
